### ISSUE TITLE
 bobuam & DotBot

### ISSUE THREAD

**Issue by eurobank** [CIDRAM/CIDRAM#529]
Hi,

DotBot is 100% respecting robots.txt and should not be blocked, i think. It is a bit overjelous but again robots.txt can block it.

User agent: Mozilla/5.0 (compatible; DotBot/1.2; +https://opensiteexplorer.org/dotbot; help@moz.com) 
Signatures reference: bobuam.php:L83 
Why blocked: Suspected bot or scraper probe! 

What do you think?

**Comment by Maikuolan** [comment]
Carefully checking over all modules, I found that Dotbot was actually being blocked by two different modules, at a total of three different places (once by BOBUAM, and twice by the Bot UAs module).

Regarding whether it *should* be blocked *at all*, different users are likely to have different views, due to it being an SEO bot (which in general, isn't something all users share the same views about, some viewing SEO bots very positively, others viewing them very negatively), and due to its requests often being unsolicited (it does seem to be correctly honouring robots.txt, which is a good thing and in its favour; there is, however, a lot of evidence for it initiating indexing for websites which don't utilise the Moz service at all, seemingly trying to index websites preemptively, even when the website owner has not explicitly requested it to do so, which while not uncommon for the average SEO service out there, is still nonetheless annoying for a lot of website owners). However, even for those that regard it as something to be blocked, having it be blocked *three times* is definitely very overzealous, and also counterproductive, as blocking it multiple times would make it more difficult to write bypasses for, and having it be easy to write bypasses for is important, for those users that regard it as something which shouldn't be blocked, and which thus would need to write bypasses for signatures which block it.

The block reason shown by BOBUAM: *"Suspected bot or scraper probe"*.
The block reason shown by the Bot UAs module: *"Backlink/SEO/Scraper UA"* and *"Banned UA"*.

I've kept the signature which shows *"Backlink/SEO/Scraper UA"* as it is, given that it most accurately and concisely describes what Dotbot is: An SEO bot. For any users wanting to bypass that signature, it should be easy enough to simply write a bypass for block reasons containing "SEO".

I've modified the other two signatures so as for them to not block Dotbot anymore (important to note that all three signatures were intended to catch a wide range of various different bots, not just Dotbot alone, so outright removing the signatures entirely wouldn't have made sense). Both of the two modified signatures, in the context of a clearly identified and well-understood bot, don't make a great deal of sense: There isn't much to be "suspected" for an already well-understood bot, and "banned UA" doesn't really tell us anything about what's being blocked, and also doesn't feel quite appropriate for any on-the-fence kind of bots (i.e., those where some users may want to block it while others may want to let it in), and feels like a more stern message to be conveying than what we're wanting here (i.e., the kind of message to convey when there's little doubt or dispute between users regarding whether something should be blocked).

In summary: BOBUAM shouldn't be blocking Dotbot anymore. The Bot UAs module will still block Dotbot, but only via *one* signature now (not two anymore), and that block can be bypassed easily for anyone needing to do so. :-)

**Comment by eurobank** [comment]
@Maikuolan They are so many SEO bots that are not blocked from CIDRAM signature and modules, so many... 

That is why i asked for DotBot to be excluded.

Also there are so many so called "security" bots, NOT respecting robots.txt that and also NOT blocked from CIDRAM.

Poor DotBot didn't need to pay the price :)

**Comment by eurobank** [comment]
Reading again your reply, i think you should revert the change. And leave the admins to whitelist whatever they want.
